### PR TITLE
Item types

### PR DESCRIPTION
**Pull request by ArmintasP**
- Changed names of private fields to follow the coding convention in `ItemType.cs`.
- Redesigned `ItemType` constructor and few methods - to retrieve or set private fields' values, each method calls property getter or setter instead of accessing the private field directly.
- Changed every use of `Enums.Category` to appropriate method or value of `ItemType.cs`

### CODE REVIEW

File: Karma/Models/Post.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ public class Post : IComparable<Post>
         [MaxLength(30)]
         public string Title { get; set; }
         [Required]
-        public Enums.Category ItemType { get; set; }
+        public int ItemType { get; set; }
```
Comment (baikauskaite) [inline Karma/Models/Post.cs:28]: I tried to create a post and not select an itemtype (left the Category field with the selection 'Select...'). Currently there's no error if you don't select an itemtype, the default value is just '0', which is an actual itemtype.

File: Karma/Models/ItemType.cs [hunk 2]
```diff
@@ -9,29 +10,28 @@ public class ItemType
     {
         public static Dictionary<int, ItemType> Types = new Dictionary<int, ItemType>();
 
-        int id;
-        string name;
-        string imagePath;
+        int _id;
+        string _name;
+        string _imagePath;
```
Comment (baikauskaite) [inline Karma/Models/ItemType.cs:15]: Shouldn't be private?

**Review by baikauskaite — APPROVED** [review]
I will also make a change in my default image picking for posts, because it doesn't work now (I used the `itemtype` enum for it).

Good work, but some more fixes will be needed in the future.

File: Karma/Models/Post.cs (hunk 1 above)
Comment (ArmintasP) [reply Karma/Models/Post.cs:28]: I'm really thankful you noticed this early on. I'm going to fix this immediately.

File: Karma/Models/ItemType.cs (hunk 2 above)
Comment (ArmintasP) [reply Karma/Models/ItemType.cs:15]: C# fields are private by default.

File: Karma/Models/Post.cs (hunk 1 above)
Comment (ArmintasP) [reply Karma/Models/Post.cs:29]: #https://github.com/PauliusLu/Pro-Bono/pull/22/commits/f2bd88d4d9ec7db5a1974d203f11e3ec04850b7b

**Pull request merged**